termwiz/src/escape/apc.rs

```diff
@@ -246,14 +246,181 @@ impl KittyImageData {
 
                 Ok(data)
             }
-            Self::SharedMem { .. } => {
-                log::error!("kitty image protocol via shared memory is not supported");
-                Err(std::io::ErrorKind::Unsupported.into())
-            }
+            Self::SharedMem {
+                name,
+                data_offset,
+                data_size,
+            } => read_shared_memory_data(&name, data_offset, data_size),
         }
     }
 }
 
+#[cfg(unix)]
+fn read_shared_memory_data(
+    name: &str,
+    data_offset: Option<u32>,
+    data_size: Option<u32>,
+) -> std::result::Result<std::vec::Vec<u8>, std::io::Error> {
+    use nix::sys::mman::{shm_open, shm_unlink};
+    use std::fs::File;
+    use std::os::unix::io::FromRawFd;
+
+    let raw_fd = shm_open(
+        name,
+        nix::fcntl::OFlag::O_RDONLY,
+        nix::sys::stat::Mode::empty(),
+    )
+    .map_err(|_| {
+        let err = std::io::Error::last_os_error();
+        std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("shm_open {} failed: {:#}", name, err),
+        )
+    })?;
+    let mut f = unsafe { File::from_raw_fd(raw_fd) };
+    if let Some(offset) = data_offset {
+        f.seek(std::io::SeekFrom::Start(offset.into()))?;
+    }
+    let data = if let Some(len) = data_size {
+        let mut res = vec![0u8; len as usize];
+        f.read_exact(&mut res)?;
+        res
+    } else {
+        let mut res = vec![];
+        f.read_to_end(&mut res)?;
+        res
+    };
+
+    if let Err(err) = shm_unlink(name) {
+        log::warn!(
+            "Unable to unlink kitty image protocol shm file {}: {:#}",
+            name,
+            err
+        );
+    }
+    Ok(data)
+}
+
+#[cfg(windows)]
+use winapi::um::{
+    handleapi::CloseHandle,
+    memoryapi::{
+        MapViewOfFile, OpenFileMappingW, UnmapViewOfFile, VirtualQuery, FILE_MAP_ALL_ACCESS,
+    },
+    winnt::{HANDLE, MEMORY_BASIC_INFORMATION},
+};
+
+#[cfg(windows)]
+struct HandleWrapper {
+    handle: HANDLE,
+}
+
+#[cfg(windows)]
+struct SharedMemObject {
+    handle: HandleWrapper,
+    buf: *mut u8,
+}
+
+#[cfg(windows)]
+impl Drop for HandleWrapper {
+    fn drop(&mut self) {
+        unsafe {
+            CloseHandle(self.handle);
+        }
+    }
+}
+
+#[cfg(windows)]
+impl Drop for SharedMemObject {
+    fn drop(&mut self) {
+        unsafe {
+            UnmapViewOfFile(self.buf as _);
+        }
+        drop(&mut self.handle);
+    }
+}
+
+#[cfg(windows)]
+fn read_shared_memory_data(
+    name: &str,
+    data_offset: Option<u32>,
+    data_size: Option<u32>,
+) -> std::result::Result<std::vec::Vec<u8>, std::io::Error> {
+    /// Convert a rust string to a windows wide string
+    fn wide_string(s: &str) -> Vec<u16> {
```

Comment:
if you add a `mod win` as suggested above, this can be extracted from here and moved into that module